### ISSUE TITLE
fix allocations in addX! in Julia 1.10

### ISSUE THREAD

**Issue by NilsNiggemann** [NilsNiggemann/PMFRG.jl#25]
The current CI is failing due to allocations in addX! that seem unrelated to this PR. 
I dont completely understand this yet but it seems that the compiler is giving up on inference at some point?
It seems that the allocations are gone if i remove @turbo which indicates that this is some compiler issue - I will try to do dig a little deeper

_Originally posted by @NilsNiggemann in https://github.com/NilsNiggemann/PMFRG.jl/issues/24#issuecomment-1896255978_

**Comment by NilsNiggemann** [comment]
There seems to be a regression in combination with LoopVectorization:

[This line]((https://github.com/NilsNiggemann/PMFRG.jl/blob/655c2b982433050bb1a6fa5a240c7c4102cce81f/src/Flowequations_Dense.jl#L447C57-L447C58)) and the ones below cause the allocations. 
The reason is that when LoopVectorization the compiler somehow cannot figure out that multiplying a float with an Int will return a float:

The minimal working example shows this:
```
using LoopVectorization
function LVTest(a1,a2)
    res = zero(eltype(a1))
    @turbo for i in eachindex(a1,a2)
        res += a1[i]*a2[i]
    end
    return res
end

aFloat = zeros(10)
aInt = zeros(Int,10)

@code_warntype LVTest(aFloat,aInt) #prints type Any for res on Julia 1.10 and does not show any type instabilities for 1.9

function checkAllocs()
    aFloat = zeros(10)
    aInt = zeros(Int,10)
    LVTest(aFloat,aInt) # compile
    println("Allocations: ",@allocated LVTest(aFloat,aInt))
end

checkAllocs() # prints 0 on Julia 1.9 but 2304 on Julia 1.10
```
The workaround is probably to explicitly convert m to a float type, but I will also file an issue in LoopVectorization

**Comment by NilsNiggemann** [comment]
fixed by [8680177](https://github.com/NilsNiggemann/PMFRG.jl/commit/8680177e6f9df26bcf9d1da3e049d3d29f7c15e1)
